library/Fission/Storage/Query/Class.hs

```diff
@@ -0,0 +1,34 @@
+module Fission.Storage.Query.Class (MonadDBQuery (..)) where
+
+import           Network.IPFS.CID.Types
+import           Database.Esqueleto hiding ((=.), update)
+
+import           Fission.Models
+import           Fission.Prelude hiding (Value)
+import           Fission.Types
+
+-- getByGeneric :: (SqlExpr (Entity model) -> SqlExpr (Value Bool)) -> Transaction m [Entity model]
```

Comment:
I tried defining a type for these functions however it seemed to rely on having access to the internals of esquelito. @expede Any ideas here?

This was the type my editor kept suggesting
```haskell
getByGeneric :: (esqueleto-3.0.0:Database.Esqueleto.Internal.Internal.SqlSelect a r, MonadIO m, BackendCompatible SqlBackend backend, PersistQueryRead backend, PersistUniqueRead backend, From a) => (a -> SqlExpr (Value Bool)) -> ReaderT backend m [r]
```

Reply:
Yeah, we can almost certainly scope this to our `Transaction` type (which only works on Postgres, but that satisfies the constraints). I'll pull it up in emacs, one sec...

Reply:
```haskell
getByGeneric :: 
  ( PersistEntity model
  , BackendCompatible SqlBackend (PersistEntityBackend model)
  , MonadIO m
  ) 
  => (SqlExpr (Entity model) -> SqlExpr (Value Bool))
  -> Transaction m [Entity model]
getByGeneric whereClause = 
  select <| from \model -> do
    where_ (whereClause model)
    return model
```

Reply:
See the other comment below for fancy protip ❇️